# HG changeset patch
# User Richard Russon <rich@flatcap.org>
# Date 1464576396 -3600
#      Mon May 30 03:46:36 2016 +0100
# Branch gentoo-1.7
# Node ID 578b9d37c9b169d369472a1b71a9d2168e4a477c
# Parent  cd990007ebab29b2529f9f37a9e7a1da4fd0c9b8
[PATCH 08/11] change_folder_next
From 5f24c40705c574d8288bf182b5482011e030f3bb Mon Sep 17 00:00:00 2001
Fabian Groffen <grobian@gentoo.org>
* This patch brings back the original behaviour of changer-folder, which
  some people find more useful.
  It suggests the next folder in your mailboxes list that has new mail
  measured from the current folder, instead of the first folder from your
  mailboxes list in any case.  This prevents starvation of folders
  defined at the back of your list ;)
  See discussion: http://thread.gmane.org/gmane.mail.mutt.devel/12457
---
 PATCHES     | 1 +
 curs_main.c | 7 +++++++
 init.h      | 7 +++++++
 mutt.h      | 1 +
 4 files changed, 16 insertions(+)

diff -Naur a/curs_main.c b/curs_main.c
--- a/curs_main.c	2016-09-03 22:22:31.941950831 -0400
+++ b/curs_main.c	2016-09-03 22:29:33.308942184 -0400
@@ -1367,7 +1367,12 @@
 #endif
 	else
 	{
+	  if (option (OPTCHANGEFOLDERNEXT) && Context && Context->path)
+	  {
+	    strfcpy (buf, Context->path, sizeof (buf));
+	    mutt_pretty_mailbox (buf, sizeof(buf));
+	  }
 	  mutt_buffy (buf, sizeof (buf));
 
           if (mutt_enter_fname (cp, buf, sizeof (buf), &menu->redraw, 1) == -1)
           {
diff -r cd990007ebab -r 578b9d37c9b1 init.h
--- a/init.h	Mon May 30 03:46:08 2016 +0100
+++ b/init.h	Mon May 30 03:46:36 2016 +0100
@@ -402,6 +402,13 @@ struct option_t MuttVars[] = {
   ** mailbox by performing a fast mailbox scan when it is defined.
   ** Afterwards the new mail status is tracked by file size changes.
   */
+  { "change_folder_next", DT_BOOL, R_NONE, OPTCHANGEFOLDERNEXT, 0 },
+  /*
+  ** .pp
+  ** When this variable is set, the change-folder command will start at the
+  ** next folder after the current folder in your mailbox list instead of
+  ** starting at the first folder.
+  */
   { "check_new",	DT_BOOL, R_NONE, OPTCHECKNEW, 1 },
   /*
   ** .pp
diff -r cd990007ebab -r 578b9d37c9b1 mutt.h
--- a/mutt.h	Mon May 30 03:46:08 2016 +0100
+++ b/mutt.h	Mon May 30 03:46:36 2016 +0100
@@ -377,6 +377,7 @@ enum
   OPTBEEP,
   OPTBEEPNEW,
   OPTBOUNCEDELIVERED,
+  OPTCHANGEFOLDERNEXT,
   OPTBRAILLEFRIENDLY,
   OPTCHECKMBOXSIZE,
   OPTCHECKNEW,
